In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import sqlite3

import warnings
warnings.filterwarnings('ignore')

In [2]:
db = 'movies.sqlite'
con = sqlite3.connect(db)
cur = con.cursor()

## QUESTIONS

#### **Question 1:** Can you get all data about movies?

In [3]:
cur.execute("select * from movies")
movies = cur.fetchall()

movies_df = pd.DataFrame(movies,columns=['Id','Original_Title','Budget','Popularity','Release_Date','Revenue','Title','Vote_Average','Vote_Count','Overview','Tagline','U_Id','Director_Id'])
print(movies_df.shape)
print('\n\n\n')
movies_df.head(3)

(4773, 13)






Id                            Original_Title     Budget  Popularity  \
0  43597                                    Avatar  237000000         150   
1  43598  Pirates of the Caribbean: At World's End  300000000         139   
2  43599                                   Spectre  245000000         107   

  Release_Date     Revenue                                     Title  \
0   2009-12-10  2787965087                                    Avatar   
1   2007-05-19   961000000  Pirates of the Caribbean: At World's End   
2   2015-10-26   880674609                                   Spectre   

   Vote_Average  Vote_Count  \
0           7.2       11800   
1           6.9        4500   
2           6.3        4466   

                                            Overview  \
0  In the 22nd century, a paraplegic Marine is di...   
1  Captain Barbossa, long believed to be dead, ha...   
2  A cryptic message from Bond’s past sends him o...   

                                          Tagline    U_Id  Director_Id  
0                     Enter the World of Pandora.   19995         4762  
1  At the end of the world, the adventure begins.     285         4763  
2                           A Plan No One Escapes  206647         4764

In [4]:
movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4773 entries, 0 to 4772
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              4773 non-null   int64  
 1   Original_Title  4773 non-null   object 
 2   Budget          4773 non-null   int64  
 3   Popularity      4773 non-null   int64  
 4   Release_Date    4773 non-null   object 
 5   Revenue         4773 non-null   int64  
 6   Title           4773 non-null   object 
 7   Vote_Average    4773 non-null   float64
 8   Vote_Count      4773 non-null   int64  
 9   Overview        4770 non-null   object 
 10  Tagline         3951 non-null   object 
 11  U_Id            4773 non-null   int64  
 12  Director_Id     4773 non-null   int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 484.9+ KB


#### **Question 2:** How do you get all data about directors?

In [5]:
cur.execute('select * from directors')
directors = cur.fetchall()
directors_df = pd.DataFrame(directors,columns=['Name','Id','Gender','U_Id','Department'])
print(directors_df.shape)
directors_df.head()

(2349, 5)


Name    Id  Gender  U_Id Department
0      James Cameron  4762       2  2710  Directing
1     Gore Verbinski  4763       2  1704  Directing
2         Sam Mendes  4764       2    39  Directing
3  Christopher Nolan  4765       2   525  Directing
4     Andrew Stanton  4766       2     7  Directing

In [6]:
directors_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2349 entries, 0 to 2348
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        2349 non-null   object
 1   Id          2349 non-null   int64 
 2   Gender      2349 non-null   int64 
 3   U_Id        2349 non-null   int64 
 4   Department  2349 non-null   object
dtypes: int64(3), object(2)
memory usage: 91.9+ KB


### Question 3: Check how many movies are present in iMDB?

In [7]:
cur.execute('select count(*) from movies')
number_of_movies = cur.fetchall()
print(number_of_movies[0])

(4773,)


### Question 4: Find these 3 directors: James Cameron ; Luc Besson ; John Woo?

In [8]:
cur.execute("select * from directors where name in ('James Cameron','Luc Besson','John Woo')")
specific_directors = cur.fetchall()
specific_directors_df = pd.DataFrame(specific_directors,columns=['Name','Id','Gender','U_Id','Department'])
specific_directors_df

Name    Id  Gender   U_Id Department
0  James Cameron  4762       2   2710  Directing
1       John Woo  4893       2  11401  Directing
2     Luc Besson  4949       2     59  Directing

### Question 5: Find all directors with name starting with Steven?

In [9]:
cur.execute("select * from directors where name like 'Steven%'")
steven_director = cur.fetchall()
steven_director_df = pd.DataFrame(steven_director,columns=['Name','Id','Gender','U_Id','Department'])
steven_director_df

Name    Id  Gender   U_Id Department
0    Steven Spielberg  4799       2    488  Directing
1   Steven Soderbergh  4909       2   1884  Directing
2        Steven Brill  5013       2  32593  Directing
3     Steven Zaillian  5117       2   2260  Directing
4        Steven Quale  5216       2  93214  Directing
5       Steven Seagal  5221       2  23880  Directing
6  Steven E. de Souza  5390       2   1726  Directing
7    Steven Shainberg  5803       2  67795  Directing
8    Steven R. Monroe  6713       2  88039  Directing

### Question 6: Count female directors?

In [10]:
cur.execute("select count(*) from directors where gender = '1'")
count_female_dir = cur.fetchall()
print(count_female_dir[0])

(150,)


### Question 7: Find the name of the 10th first women directors?

In [11]:
cur.execute("select name from directors where gender = '1'order by id limit 1 offset 10 ")
name_10th_female_dir = cur.fetchall()
print(name_10th_female_dir[0])

('Karyn Kusama',)


### Question 8: What are the 3 most popular movies?

In [12]:
cur.execute("select original_title from movies order by popularity desc limit 3")
three_popular_movies = cur.fetchall()

print(three_popular_movies)

[('Minions',), ('Interstellar',), ('Deadpool',)]


### Question 9: What are the 3 most bankable movies?

In [13]:
cur.execute("select original_title from movies order by budget desc limit 3")
three_bankable_movies = cur.fetchall()

print(three_bankable_movies)

[('Pirates of the Caribbean: On Stranger Tides',), ("Pirates of the Caribbean: At World's End",), ('Avengers: Age of Ultron',)]


### Question 10: What is the most awarded average vote since the January 1st, 2000?

In [14]:
cur.execute("select original_title from movies where release_date > '2000-01-01' order by vote_average desc limit 1")
most_awarded_avg_vote = cur.fetchall()

print(most_awarded_avg_vote)

[('Sardaarji',)]


### Question 11: Which movie(s) were directed by Brenda Chapman?

In [15]:
cur.execute("select d.Name,m.original_title from movies m inner join directors d on d.Id = m.Director_Id where d.Name = 'Brenda Chapman'")
brenda_movies = cur.fetchall()

print(brenda_movies)

[('Brenda Chapman', 'Brave')]


### Question 12: Whose director made the most movies?

In [16]:
cur.execute("select d.Name,count(m.original_title) from movies m inner join directors d on m.Director_Id = d.id group by d.Name order by count(m.original_title) desc limit 1")
director_with_maximum_movies = cur.fetchall()

print(director_with_maximum_movies)

[('Steven Spielberg', 27)]


### Question 13: Whose director is the most bankable?

In [17]:
cur.execute("select d.name,sum(m.budget) from movies m inner join directors d on m.Director_Id = d.Id group by d.Name order by sum(m.budget) desc limit 1")
most_bankable_director = cur.fetchall()

print(most_bankable_director)

[('Steven Spielberg', 1667500000)]


# Analysis on IMDB DATA SET

### Budget Analysis

**Question** - List out Top 10 highest budget made movies 

In [18]:
cur.execute('SELECT original_title, name, release_date, budget, revenue FROM movies JOIN directors ON movies.director_id = directors.id ORDER BY budget DESC')
mostExpensive = cur.fetchall()
mostExpensive = pd.DataFrame(mostExpensive, columns = ['original_title', 'director_name', 'release_date', 'budget', 'revenue'])
mostExpensive.head(10)

original_title      director_name  \
0  Pirates of the Caribbean: On Stranger Tides       Rob Marshall   
1     Pirates of the Caribbean: At World's End     Gore Verbinski   
2                      Avengers: Age of Ultron        Joss Whedon   
3                             Superman Returns       Bryan Singer   
4                                  John Carter     Andrew Stanton   
5                                      Tangled       Byron Howard   
6                                 Spider-Man 3          Sam Raimi   
7                              The Lone Ranger     Gore Verbinski   
8                        The Dark Knight Rises  Christopher Nolan   
9       Harry Potter and the Half-Blood Prince        David Yates   

  release_date     budget     revenue  
0   2011-05-14  380000000  1045713802  
1   2007-05-19  300000000   961000000  
2   2015-04-22  280000000  1405403694  
3   2006-06-28  270000000   391081192  
4   2012-03-07  260000000   284139100  
5   2010-11-24  260000000   591794936  
6   2007-05-01  258000000   890871626  
7   2013-07-03  255000000    89289910  
8   2012-07-16  250000000  1084939099  
9   2009-07-07  250000000   933959197


> ## Note :

The mostExpensive Table Contains List Of Movies With Highest Production Costs. 

For Your Information, Pirates of the Caribbean: On Stranger Tides, Directed By Rob Marshall, Is The Most Production Costs, It's USD 380,000,000.

### Popularity Analysis

**Question**- List out Top 10 popularity on movies

In [19]:
cur.execute("select original_title,Name,release_date,popularity from movies m inner join directors d on m.director_id = d.id order by popularity desc")
most_popular_movies = cur.fetchall()

most_popular_movies_df = pd.DataFrame(most_popular_movies,columns=['Movie','Director_Name','Release_Date','Popularity'])
most_popular_movies_df.head(10)

Movie      Director_Name  \
0                                            Minions         Kyle Balda   
1                                       Interstellar  Christopher Nolan   
2                                           Deadpool         Tim Miller   
3                            Guardians of the Galaxy         James Gunn   
4                                 Mad Max: Fury Road      George Miller   
5                                     Jurassic World    Colin Trevorrow   
6  Pirates of the Caribbean: The Curse of the Bla...     Gore Verbinski   
7                     Dawn of the Planet of the Apes        Matt Reeves   
8              The Hunger Games: Mockingjay - Part 1   Francis Lawrence   
9                                         Big Hero 6     Chris Williams   

  Release_Date  Popularity  
0   2015-06-17         875  
1   2014-11-05         724  
2   2016-02-09         514  
3   2014-07-30         481  
4   2015-05-13         434  
5   2015-06-09         418  
6   2003-07-09         271  
7   2014-06-26         243  
8   2014-11-18         206  
9   2014-10-24         203

> ## Note:

most_popular_movies_df Table Shows 10 Movies With The Highest Ranking By Popularity. 

Currently, Minions Directed By Kyle Balda Got The First Place With Score 875, Beating Interstellar As The Closest Competitor, Deadpool And The Other Competitors.

### Revenu Analysis

**Question** - What are the top 10 Revenue  movies  

In [21]:
cur.execute('select original_title, name, release_date, revenue from movies m join directors d on d.id = m.Director_id order by revenue desc')
top_10_revenue = cur.fetchall()
top_10_revenue_df = pd.DataFrame(top_10_revenue,columns=['Original_Title','Name','Release_Date','Revenue'])
top_10_revenue_df.head(10)

Original_Title             Name Release_Date     Revenue
0                      Avatar    James Cameron   2009-12-10  2787965087
1                     Titanic    James Cameron   1997-11-18  1845034188
2                The Avengers      Joss Whedon   2012-04-25  1519557910
3              Jurassic World  Colin Trevorrow   2015-06-09  1513528810
4                   Furious 7        James Wan   2015-04-01  1506249360
5     Avengers: Age of Ultron      Joss Whedon   2015-04-22  1405403694
6                      Frozen       Chris Buck   2013-11-27  1274219009
7                  Iron Man 3      Shane Black   2013-04-18  1215439994
8                     Minions       Kyle Balda   2015-06-17  1156730962
9  Captain America: Civil War    Anthony Russo   2016-04-27  1153304495

> ## Note :

The mostProfit Table Shows The 10 Movies Got The Highest Gross Profit When Compared To The Other Movies.

It Turns Out That Avatar Directed By James Cameron Got A Gross Profit USD 2,787,965,087. It's Make Avatar In First Place, Followed By Titanic, The Avengers, Jurassic World, And Others.

From The Table, It Can Be Seen That James Cameron Make Two Movies To Fill The First And Second Place On The Rankings.

**Question** - Display most popular movies with vote_average

In [29]:
cur.execute('select original_title,name,release_date,budget,revenue,popularity,vote_average from directors d join movies m on d.id = m.director_id where name = "James Cameron" order by revenue desc')
most_popular_movies_revenue = cur.fetchall()
most_popular_movies_revenue_df = pd.DataFrame(most_popular_movies_revenue,columns=['Original_Title','name','Release_Date','Budget','Revenue','Popularity','Vote_Average'])
most_popular_movies_revenue_df

Original_Title           name Release_Date     Budget  \
0                      Avatar  James Cameron   2009-12-10  237000000   
1                     Titanic  James Cameron   1997-11-18  200000000   
2  Terminator 2: Judgment Day  James Cameron   1991-07-01  100000000   
3                   True Lies  James Cameron   1994-07-14  115000000   
4                      Aliens  James Cameron   1986-07-18   18500000   
5                   The Abyss  James Cameron   1989-08-09   70000000   
6              The Terminator  James Cameron   1984-10-26    6400000   

      Revenue  Popularity  Vote_Average  
0  2787965087         150           7.2  
1  1845034188         100           7.5  
2   520000000         101           7.7  
3   378882411          38           6.8  
4   183316455          67           7.7  
5    90000098          24           7.1  
6    78371200          74           7.3

> ## Note : 

From the mostProfit Table, It Can Be Seen That The Movies In The First And Second Rank Is A Movies Directed By James Cameron.

In The Table moviesByJamesCameron Shows All Movies Directed By James Cameron.

### Voting Analysis

**Question** - List out Particular movie for Voting average and votecount 

In [40]:
cur.execute('select original_title,name,release_date,vote_average,vote_count from directors d join movies m on d.id = m.director_id order by vote_average desc')
bestvote = cur.fetchall()
bestvote_df = pd.DataFrame(bestvote,columns=['Original_Title','Name','Release_Date','Vote_Average','Vote_Count'])
bestvote_df.head(10)

Original_Title                  Name Release_Date  Vote_Average  \
0          Stiff Upper Lips           Gary Sinyor   1998-06-12          10.0   
1     Dancer, Texas Pop. 81         Tim McCanlies   1998-05-01          10.0   
2                 Sardaarji          Rohit Jugraj   2015-06-26           9.5   
3            One Man's Hero            Lance Hool   1999-08-02           9.3   
4  The Shawshank Redemption        Frank Darabont   1994-09-23           8.5   
5        There Goes My Baby          Floyd Mutrux   1994-09-02           8.5   
6     The Prisoner of Zenda         John Cromwell   1937-09-03           8.4   
7             The Godfather  Francis Ford Coppola   1972-03-14           8.4   
8                Fight Club         David Fincher   1999-10-15           8.3   
9          Schindler's List      Steven Spielberg   1993-11-29           8.3   

   Vote_Count  
0           1  
1           1  
2           2  
3           2  
4        8205  
5           2  
6          11  
7        5893  
8        9413  
9        4329

> ## Note :

BestVote Table Contains The Best Movies By Vote. Stiff Upper Lips, Directed By Gary Sinyor And Dancer, Texas Pop. 81 Directed by the MacCanlies Was Successfully Becomes the Best Movies By Vote With Perfect Score, 10.

However, There Are Anomalies. It Turned Out That The Two Movies Were Only Vote By One Person. So, Likely to Affect The Assessment Results..

### Directors

**Question**- List out the  director names with number of movies and revenue

In [44]:
cur.execute('select name,count(original_title),sum(revenue) from directors d join movies m on d.id = m.director_id group by name order by sum(revenue) desc')
director_name_Movie = cur.fetchall()
director_name_Movie_df = pd.DataFrame(director_name_Movie,columns=['Name','Number_Movies','Sum_Revenue'])
director_name_Movie_df

Name  Number_Movies  Sum_Revenue
0      Steven Spielberg             27   9147393164
1         Peter Jackson              9   6498642820
2         James Cameron              7   5883569439
3           Michael Bay             12   5832524638
4     Christopher Nolan              8   4227483234
...                 ...            ...          ...
2344         Adam Green              1            0
2345      Adam Goldberg              1            0
2346       Adam Carolla              1            0
2347    Aaron Schneider              1            0
2348         Aaron Hann              1            0

[2349 rows x 3 columns]



> ## Note :

The mostProfitableDirector Table Shows The Directors With The Highest Gross Profit. 

Currently, Steven Spielberg Is The Director With The Most Gross Profit, with USD 9,147,393,164 from his 27 movies. 

Then, Peter Jackson With USD 6,498,642,820, James Cameron With USD 5,883,569,439, And The Other Directors.

**Question** - Display all the number of movies for particular director and revenue .

In [45]:
cur.execute('select name,count(original_title),sum(revenue) from directors d join movies m on d.id = m.director_id group by name order by count(original_title) desc')
all_movies_dir = cur.fetchall()
all_movies_dir_df = pd.DataFrame(all_movies_dir,columns = ['Name','Movies','Revenue'])
all_movies_dir_df

Name  Movies     Revenue
0     Steven Spielberg      27  9147393164
1          Woody Allen      21   669101038
2      Martin Scorsese      20  1956635998
3       Clint Eastwood      20  2512058888
4            Spike Lee      16   340618771
...                ...     ...         ...
2344      Adam Carolla       1           0
2345       Adam Brooks       1    55447968
2346      Abel Ferrara       1     1227324
2347   Aaron Schneider       1           0
2348        Aaron Hann       1           0

[2349 rows x 3 columns]


> ### Note :

The mostProductiveDirector Table Contains The Name Of The Director And The Number Of Movies It Has Directed By Him. 

Steven Spielberg Is The Most Productive Director With 27 Movies Titles He Has Directed. Meanwhile, Woody Allen Has Only 21 Movies Titles. 

Then, Martin Scorsese And Clint Eastwood With 20 Movies Titles.

**Question**- By doing Director analysis We know Steven Spielberg is the highest revenue, so list the Steven Spielberg movies info

In [46]:
cur.execute('select original_title, release_date, budget, revenue, popularity, vote_average from directors d join movies m on d.id = m.director_id where name = "Steven Spielberg" order by release_date desc')
steven_movies = cur.fetchall()
steven_movies_df = pd.DataFrame(steven_movies,columns=['Title','Release_Date','Budget','Revenue','Popularity','Vote_Average'])
steven_movies_df

Title Release_Date     Budget  \
0                                             The BFG   2016-06-01  140000000   
1                                     Bridge of Spies   2015-10-15   40000000   
2                                             Lincoln   2012-11-09   65000000   
3                                           War Horse   2011-12-25   66000000   
4                            The Adventures of Tintin   2011-10-25  130000000   
5   Indiana Jones and the Kingdom of the Crystal S...   2008-05-21  185000000   
6                                              Munich   2005-12-22   70000000   
7                                   War of the Worlds   2005-06-28  132000000   
8                                        The Terminal   2004-06-17   60000000   
9                                 Catch Me If You Can   2002-12-25   52000000   
10                                    Minority Report   2002-06-20  102000000   
11                       A.I. Artificial Intelligence   2001-06-29  100000000   
12                                Saving Private Ryan   1998-07-24   70000000   
13                                            Amistad   1997-12-03   36000000   
14                      The Lost World: Jurassic Park   1997-05-23   73000000   
15                                   Schindler's List   1993-11-29   22000000   
16                                      Jurassic Park   1993-06-11   63000000   
17                                               Hook   1991-12-11   70000000   
18                 Indiana Jones and the Last Crusade   1989-05-24   48000000   
19                                   The Color Purple   1985-12-18   15000000   
20               Indiana Jones and the Temple of Doom   1984-05-23   28000000   
21                           Twilight Zone: The Movie   1983-06-24   10000000   
22                         E.T. the Extra-Terrestrial   1982-04-03   10500000   
23                            Raiders of the Lost Ark   1981-06-12   18000000   
24                                               1941   1979-12-13   35000000   
25                 Close Encounters of the Third Kind   1977-11-16   20000000   
26                                               Jaws   1975-06-18    7000000   

      Revenue  Popularity  Vote_Average  
0   183345589          44           6.0  
1   165478348          48           7.2  
2   275293450          36           6.7  
3   177584879          29           7.0  
4   371940071          89           6.7  
5   786636033          75           5.7  
6   130358911          29           6.9  
7   591739379          48           6.2  
8   219417255          57           7.0  
9   352114312          73           7.7  
10  358372926          65           7.1  
11  235926552          34           6.8  
12  481840909          76           7.9  
13   74000000           3           6.8  
14  229074524           2           6.2  
15  321365567         104           8.3  
16  920100000          40           7.6  
17  300854823          33           6.6  
18  474171806          80           7.6  
19  146292009          17           7.7  
20  333000000          66           7.1  
21   29450919          12           6.2  
22  792910554          56           7.3  
23  389925971          68           7.7  
24   31755742          10           5.6  
25  303788635          52           7.2  
26  470654000          50           7.5

> ## Note : 

From Two Tables Above, mostProfitableDirector And mostProductiveDirector, It Can Be Seen That Steven Spielberg Is Always In The First Place. 

It Can Be Understood That Steven Spielberg Has Produced Many And Incredible Movies That Are Shows In The Table MoviesByStevenSpielberg Above.
